types/mapbox-gl/index.d.ts

```diff
@@ -315,7 +315,7 @@ declare namespace mapboxgl {
 
         getLayer(id: string): mapboxgl.Layer;
 
-        setFilter(layer: string, filter?: any[] | boolean | null): this;
+        setFilter(layer: string, filter?: any[] | boolean | null, options?: { validate?: boolean }): this;
```

Comment:
Technically, we should also allow `null`s here, as that is Flow semantic for optional values:

```
setFilter(layer: string, filter?: any[] | boolean | null, options?: { validate?: boolean | null} | null): this;
```

I'd also prefer to define SetFilterOptions interface instead of an inline definition

Reply:
Agreed 👍 

I've ended up calling it `FilterOptions`, and have also applied this interface to the `queryRenderedFeatures` & `querySourceFeatures`. Hope that works for you!